### PR TITLE
Implement UI for unavailable items

### PR DESCRIPTION
**Pull request by marcoskwkm**
#### What problem is this solving?

This PR:
- Displays unavailable and available items separately;
- Displays a message explaining why the item is unavailable.

#### How should this be manually tested?

[Use this workspace](https://unavailable--vtexgame1.myvtex.com/cart).
Add [this item](https://unavailable--vtexgame1.myvtex.com/teste-so-delivery-copy-256--copy-257--copy-261--copy-262-/p) and [this item](https://unavailable--vtexgame1.myvtex.com/produto-variacao/p?skuId=14) to your cart, proceed to the _**old**_ checkout (`/checkout`) and set delivery to somewhere in SP (you can use 04538-132). This makes the second item unavailable. Now proceed to `/cart` and verify the product-list behaves as expected.

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [x] [Linked this PR to a Clubhouse story (if applicable)](https://app.clubhouse.io/vtex/story/18328/implementar-ui-dos-erros-de-disponibilidade-product-list).
- [x] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Screenshots or example usage
Screen Shot 2019-09-06 at 3 29 33 PM

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

**Review by jeffersontuc — APPROVED** [review]
lgtm

File: react/components/ListItem.tsx [hunk 1]
```diff
@@ -14,130 +16,152 @@ interface Props {
...
-          />
-        </div>
+          {/* Quantity Selector */}
+          <div className={`${styles.quantity} flex-none-m`}>
+            <div className={`${styles.quantitySelector}`}>
+              <Selector
+                value={item.quantity}
+                maxValue={MAX_ITEM_QUANTITY}
+                onChange={onQuantityChange}
+                disabled={item.availability !== 'available'}
```
Comment (darlenedms) [inline react/components/ListItem.tsx]: You've created a constant for this but it's not using it.

File: react/components/ListItem.tsx [hunk 2]
```diff
@@ -14,130 +16,152 @@ interface Props {
...
+                className="pointer bg-transparent bn pa2 ml6"
+                onClick={onRemove}
+              >
+                <IconDelete color="#727273" />
+              </button>
+            </div>
+          </div>
         </div>
+
+        {item.availability !== 'available' ? (
```
Comment (darlenedms) [inline react/components/ListItem.tsx]: Ditto.

File: react/ProductList.tsx [hunk 3]
```diff
@@ -1,6 +1,8 @@
 import React, { FunctionComponent } from 'react'
+import { FormattedMessage } from 'react-intl'
 
 import ListItem from './components/ListItem'
+import Availability from './constants/Availability'
```
Comment (darlenedms) [inline react/ProductList.tsx]: ⛏ I would import `AVAILABLE` directly, I feel it's easier to read.

Comment (marcoskwkm) [reply react/ProductList.tsx]: Fair point.

File: react/components/ListItem.tsx (hunk 1 above)
Comment (marcoskwkm) [reply react/components/ListItem.tsx]: Oops, missed this.

**Review by darlenedms — APPROVED** [review]

**Pull request merged**